src/Fields/fieldvector.jl

```diff
@@ -184,6 +185,44 @@ end
     return dest
 end
 
+"""
+    Spaces.create_dss_buffer(fv::FieldVector)
+
+Create a NamedTuple of buffers for communicating neighbour information of
+each Field in `fv`. In this NamedTuple, the name of each field is mapped
+to the buffer.
+"""
+function Spaces.create_dss_buffer(fv::FieldVector)
+    NamedTuple{propertynames(fv)}(
+        unrolled_map(
+            key -> Spaces.create_dss_buffer(getproperty(fv, key)),
+            propertynames(fv),
+        ),
+    )
```

Comment:
Is `unrolled_map` actually needed here? My preference would be to use `map` until we deem `unrolled_map` needed.

Reply:
In ClimaLand, our state is a FieldVector with Fields defined on different spaces - would this count as a nonuniform collection that we'd want to use `unrolled_map` with?

Reply:
`unrolled_map` is just a optimized form of `Base`'s `map` to overcome limitations with compiler inference. So, `map` may result in inference failures, but it should work just fine. Using `unrolled_map` is not critical here in the sense that `create_dss_buffer` is only called during initialization, and it's not called inside a gpu kernel, so the only penalty is that `map` _could_ lead to runtime dispatch.

If we found that this fixes inference failures in this function, then `unrolled_map` is fine, but it's not strictly needed (as it is in other places).

Reply:
Thanks for the explanation! I'll change it to `map`